microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthCdiExtension.java

```diff
@@ -139,13 +150,47 @@ public void registerClaimProducers(@Observes AfterBeanDiscovery abd, BeanManager
      *
      * @param add event from CDI container
      */
-    public void validate(@Observes AfterDeploymentValidation add) {
+    void validate(@Observes AfterDeploymentValidation add) {
         qualifiers.forEach(q -> {
             ClaimLiteral claimLiteral = q.getQualifier();
             validate(claimLiteral);
         });
     }
 
+    void configured(@Observes @RuntimeStart Config config) {
+        this.config = config;
+    }
+
+    void registerProvider(@Observes
+                          @Initialized(ApplicationScoped.class)
+                          @Priority(PLATFORM_BEFORE + 5) Object event,
+                          BeanManager bm) {
+        // I need the JAX-RS extension and the Security extension
+        JaxRsCdiExtension jaxrs = bm.getExtension(JaxRsCdiExtension.class);
+        SecurityCdiExtension security = bm.getExtension(SecurityCdiExtension.class);
+
+        boolean notNeeded = jaxrs.applicationsToRun()
+                .stream()
+                .map(JaxRsApplication::applicationClass)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(clazz -> clazz.getAnnotation(LoginConfig.class))
+                .filter(Objects::nonNull)
+                .map(LoginConfig::authMethod)
+                .noneMatch("MP-JWT"::equals);
+
+        if (notNeeded) {
+            return;
+        }
+
+        if (security.securityBuilder().hasProvider("mp-jwt-auth")) {
+            return;
+        }
```

Comment:
I would say, that this check should be right after obtaining security. That mean somewhere around line 171.